tuyaha/devices/switch.py

```diff
@@ -1,10 +1,14 @@
 from tuyaha.devices.base import TuyaDevice
 
-
 class TuyaSwitch(TuyaDevice):
 
     def state(self):
-        state = self.data.get('state')
+        devices = self.api.discover_devices()
```

Comment:
This does a request which now causes home assistant event loop to block.

Home assistant needs to be updated to not call this method `state` in the entity `is_on` property. It should be called in `update` entity method instead and cache the result.